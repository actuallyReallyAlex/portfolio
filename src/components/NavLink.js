import React from 'react'
import PropTypes from 'prop-types'
import { Anchor } from 'grommet'
import { smoothlyScroll } from '../util'
import { Link } from 'react-router-dom'

const DefaultLink = ({ active, title }) => (
  <Anchor
    color="black"
    className={`nav-link slow-transition ${active ? 'active' : ''}`}
    onClick={() => smoothlyScroll(title)}
  >
    {title}
  </Anchor>
)

const dict = {
  Home: {
    to: '/'
  },
  Blog: {
    to: '/blog'
  }
}

const RouterLink = ({ active, title }) => (
  <Link to={dict[title].to}>{title}</Link>
)

const NavLink = ({ active, title }) =>
  title === 'Home' || title === 'Blog' ? (
    <RouterLink active={active} title={title} />
  ) : (
    <DefaultLink active={active} title={title} />
  )

NavLink.propTypes = {
  active: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired
}

export default NavLink
