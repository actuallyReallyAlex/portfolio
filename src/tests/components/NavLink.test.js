import React from 'react'
import NavLink from '../../components/NavLink'
import { shallow } from 'enzyme'

const defaultProps = {
  title: 'Title'
}

describe('<NavLink />', () => {
  afterEach(() => {
    // defaultProps.handler.mockClear()
  })

  test('Should render the <NavLink /> component.', () => {
    const component = shallow(<NavLink {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })

  test('Should smoothly scroll to element on link click.', () => {
    const component = shallow(<NavLink {...defaultProps} />)
    component.simulate('click')
  })
})
