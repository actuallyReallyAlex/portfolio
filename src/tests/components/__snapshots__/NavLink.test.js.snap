// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLink /> Should render the <NavLink /> component. 1`] = `ShallowWrapper {}`;
